File: packages/python/runfile_ai/policy.py
"""Redaction-policy fetch + cache.

Fetched from ``GET /v1/policies/current`` on startup and refreshed every TTL
(default 300s) in the background. Configures the classifier/redactor. On fetch
failure the SDK keeps using the cached policy and emits an ``sdk_diagnostic``.
Each captured item records the ``redaction_policy_version`` active at capture.

Skeleton: signatures in place; bodies TODO.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any

DEFAULT_TTL_SECONDS = 300


@dataclass
class RedactionPolicy:
    policy_version: str
    classification_rules: list[dict[str, Any]]
    ttl_seconds: int = DEFAULT_TTL_SECONDS


class PolicyCache:
    def __init__(self, *, ttl_seconds: int = DEFAULT_TTL_SECONDS) -> None:
        self._ttl = ttl_seconds

    async def get(self) -> RedactionPolicy:
        raise NotImplementedError

    async def refresh(self) -> RedactionPolicy:
        raise NotImplementedError

File: packages/typescript/src/policy.ts
/**
 * Redaction-policy fetch + cache.
 *
 * Fetched from `GET /v1/policies/current` on startup and refreshed every TTL
 * (default 300s) in the background. Configures the classifier/redactor. On fetch
 * failure the SDK keeps using the cached policy and emits an `sdk_diagnostic`.
 *
 * Skeleton: signatures in place; bodies TODO.
 */

export const DEFAULT_TTL_SECONDS = 300;

export interface RedactionPolicy {
  policyVersion: string;
  classificationRules: Array<Record<string, unknown>>;
  ttlSeconds: number;
}

export class PolicyCache {
  constructor(private readonly ttlSeconds: number = DEFAULT_TTL_SECONDS) {}

  async get(): Promise<RedactionPolicy> {
    throw new Error('not implemented');
  }

  async refresh(): Promise<RedactionPolicy> {
    throw new Error('not implemented');
  }
}
